### PR TITLE
[DPE-1676] Allow scale down to 0 nodes

### PR DESCRIPTION
**Pull request by Mehdi-Bendriss**
## Issue
This PR implements [DPE-1676](https://warthogs.atlassian.net/browse/DPE-1676) - namely, this PR implements:
- removal of blocking safeguards when scale-down to 0 nodes 
- improvement of locking logic on scale-down

[DPE-1676]: https://warthogs.atlassian.net/browse/DPE-1676?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/charms/opensearch/v0/opensearch_locking.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    def acquire(self):
+        """Method for Acquiring the "ops" lock."""
+        # no lock acquisition needed if only 1 unit remaining
+        if len(self._charm.model.get_relation(PeerRelationName).units) == 1:
+            return
+
+        host = self._charm.unit_ip if self._opensearch.is_node_up() else None
+
+        # can use opensearch to lock
+        if host is not None or self._charm.alt_hosts:
```
Comment (delgod) [inline lib/charms/opensearch/v0/opensearch_locking.py:46]: Should we try lock via OpenSearch if lock exists in peer databag?

**Review by delgod — APPROVED** [review]

File: lib/charms/opensearch/v0/opensearch_locking.py (hunk 1 above)
Comment (Mehdi-Bendriss) [reply lib/charms/opensearch/v0/opensearch_locking.py:50]: Good point, thank you

**Review by delgod — APPROVED** [review]

**Review by WRFitch — APPROVED** [review]
nice! Do we still need those checks in the HA tests that verify we aren't scaling down past 1 unit now?

**Comment by Mehdi-Bendriss** [comment]
@WRFitch - yes, those tests must run against a healthy cluster (+1 node)

**Pull request merged**